### PR TITLE
tapfreighter: validate proof courier address before commencing send

### PR DESCRIPTION
**Pull request by ffranr**
This PR closes issue https://github.com/lightninglabs/taproot-assets/issues/455 .

### CODE REVIEW

File: tapfreighter/chain_porter.go [hunk 1]
```diff
@@ -1030,6 +1038,24 @@ func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
 	}
 }
 
+// validateAddrParcel validates the given address parcel.
+func (p *ChainPorter) validateAddrParcel(addrParcel AddressParcel) error {
+	for idx := range addrParcel.destAddrs {
```
Comment (positiveblue) [inline tapfreighter/chain_porter.go:1043]: the check for empty `destAddrs` is done before getting here right?

**Comment by ffranr** [comment]
@positiveblue Thanks for the review!

I think I've found a better methodology for this change. I've updated the PR, apologies for the late change.

**Review by positiveblue — APPROVED** [review]
I like the new approach more, we could incorporate something like this in our rpc flows too :100:

**Review by Roasbeef — APPROVED** [review]
LGTM 💣

**Pull request merged**